Sources/Brave/Frontend/Browser/PageData.swift

```diff
@@ -98,18 +98,26 @@ struct PageData {
       }
     }
     
+    let allEngineScriptTypes = await makeAllEngineScripts(for: domain)
+    return userScriptTypes.union(allEngineScriptTypes)
+  }
+  
+  func makeMainFrameEngineScriptTypes(domain: Domain) async -> Set<UserScriptType> {
+    return await adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
+  }
+  
+  func makeAllEngineScripts(for domain: Domain) async -> Set<UserScriptType> {
     // Add engine scripts for the main frame
-    userScriptTypes = userScriptTypes.union(
-      adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
-    )
+    async let engineScripts = adBlockStats.makeEngineScriptTypes(frameURL: mainFrameURL, isMainFrame: true, domain: domain)
     
     // Add engine scripts for all of the known sub-frames
-    let additionalScriptTypes = allSubframeURLs.map({ frameURL in
-      return self.adBlockStats.makeEngineScriptTypes(frameURL: frameURL, isMainFrame: false, domain: domain)
+    async let additionalScriptTypes = allSubframeURLs.asyncConcurrentCompactMap({ frameURL in
+      return await self.adBlockStats.makeEngineScriptTypes(frameURL: frameURL, isMainFrame: false, domain: domain)
```

Comment:
This `asyncConcurrentCompactMap` is completely useless as `makeEngineScriptTypes` is `MainActor` isolated thus won't execute concurrently

Reply:
my next pr fixes this, I was hoping to keep it to have less merge conflicts as I borrowed a subset of changes. from that pr.

Reply:
No sorry, fix it here. If the contents inside need to be async then use `asyncCompactMap` instead

Reply:
I can get rid of the @MainActor actually its only needed because of `cachedEngine.isEnabled(for: domain)` but I can just make this await.